### PR TITLE
Add TimeWindow reader

### PR DESCRIPTION
**Pull request by junjiejiangjjj**
Signed-off-by: junjiejiangjjj <junjie.jiang@zilliz.com>

### CODE REVIEW

File: towhee/engine/operator_io/reader.py [hunk 1]
```diff
@@ -136,3 +136,105 @@ def read(self) -> List[Dict[str, any]]:
...
         self._close = True
         self._iter.notify()
+
+
+class _TimeWindow:
+    '''
+    '''
+
+    def __init__(self, time_range_sec: int, time_step_sec: int, start_time_sec: int = 0):
+        self._start_time_m = start_time_sec * 1000
```
Comment (filip-halt) [inline towhee/engine/operator_io/reader.py:146]: Why are the seconds multiplied by 1000 throughout the code?

File: towhee/engine/operator_io/reader.py [hunk 2]
```diff
@@ -136,3 +136,105 @@ def read(self) -> List[Dict[str, any]]:
...
+
+            while True:
+                if self._window is None:
+                    raise StopIteration
+
+                try:
+                    data = next(self._iter)
+                except StopIteration:
+                    if self._window is not None:
+                        ret = [self._to_op_inputs(row) for row in self._window.data]
```
Comment (filip-halt) [inline towhee/engine/operator_io/reader.py:222]: when are we setting self._window.data?

File: towhee/engine/operator_io/reader.py (hunk 1 above)
Comment (junjiejiangjjj) [reply towhee/engine/operator_io/reader.py:149]: The timestamp of video frame is millisecond.

File: towhee/engine/operator_io/reader.py (hunk 2 above)
Comment (junjiejiangjjj) [reply towhee/engine/operator_io/reader.py]: In `_TimeWindow` class

**Pull request merged**